validation/schema_testing/officers/officerDeltaSchema_test.go

```diff
@@ -0,0 +1,198 @@
+
+	Convey("Given I want to test the officers-delta API schema to assert mandatory validation is working correctly", t, func() {
+
+		mandatoryErrorsRequestBody := readRequestBody(requiredErrorRequestBodyLocation)
+
+		r := httptest.NewRequest("POST", officersEndpoint, bytes.NewBuffer(mandatoryErrorsRequestBody))
+		r.Header.Set("X-Request-Id", contextId)
+		r.Header.Set("Content-Type", "application/json")
+
+		Convey("When I call to validate the request body, providing an valid request with missing mandatory values", func() {
+
+			chv := validation.NewCHValidator()
+
+			validationErrs, _ := chv.ValidateRequestAgainstOpenApiSpec(r, apiSpecLocation, contextId)
+
+			Convey("Then I am given an errors array response as validation errors have been found", func() {
+				mandatoryErrorsResponseBody := readRequestBody(requiredErrorResponseBodyLocation)
+				match := compareActualToExpected(validationErrs, mandatoryErrorsResponseBody)
+
+				So(validationErrs, ShouldNotBeNil)
+				So(match, ShouldEqual, true)
+			})
+		})
+	})
+}
+
+// TestOfficerDeltaSchemaEnumErrors asserts that when an invalid request body is given with incorrect ENUM values.
+// then an errors array is returned, stating that given values are incorrect.
+func TestOfficerDeltaSchemaEnumErrors(t *testing.T) {
+
+	Convey("Given I want to test the officers-delta API schema to assert ENUM validation is working correctly", t, func() {
+
+		enumErrorsRequestBody := readRequestBody(enumErrorRequestBodyLocation)
+
+		r := httptest.NewRequest("POST", officersEndpoint, bytes.NewBuffer(enumErrorsRequestBody))
+		r.Header.Set("X-Request-Id", contextId)
+		r.Header.Set("Content-Type", "application/json")
+
+		Convey("When I call to validate the request body, providing an valid request with incorrect ENUM values", func() {
+
+			chv := validation.NewCHValidator()
+
+			validationErrs, _ := chv.ValidateRequestAgainstOpenApiSpec(r, apiSpecLocation, contextId)
+
+			Convey("Then I am given an errors array response as validation errors have been found", func() {
+				enumErrorsResponseBody := readRequestBody(enumErrorResponseBodyLocation)
+				match := compareActualToExpected(validationErrs, enumErrorsResponseBody)
+
+				So(validationErrs, ShouldNotBeNil)
+				So(match, ShouldEqual, true)
+			})
+		})
+	})
+}
+
+// readRequestBody takes a file location in the format of a string relative path and reads in the contents from the file,
+// removing any extra tabs, spaces and special characters using json.Compact. Returns a []byte version of the formatted file read in.
+func readRequestBody(fl string) []byte {
+
+	// Read in contents and covert it to a string for further processing.
+	raw, _ := os.ReadFile(fl)
+
+	buffer := new(bytes.Buffer)
+	_ = json.Compact(buffer, raw)
+
+	raw = buffer.Bytes()
+	// Convert back to an []byte and return.
+	return raw
+}
+
+// compareActualToExpected takes an actual and expected byte array and compares them to see if they match. Ordering of response
```

Comment:
Perhaps "takes actual and expected json (as byte arrays) and compares them to see if they match

Reply:
Agreed, this would describe it a bit clearer. Will change to the suggestion